src/main/java/com/labsynch/labseer/api/ApiParentController.java

```diff
@@ -67,6 +74,31 @@ public ResponseEntity<String> updateParent(@RequestBody String json) {
 		}
 	}
 
+	@Transactional
+	@RequestMapping(value = "/swapParentStructures", method = RequestMethod.POST, headers = "Accept=application/json")
+	@ResponseBody
+	public ResponseEntity<String> swapParentStructures(@RequestBody String json) {
+		HttpHeaders headers = new HttpHeaders();
+		ParentSwapStructuresDTO parentSwapStructuresDTO = ParentSwapStructuresDTO.fromJSONToParentSwapStructuresDTO(json);
+
+		String msg = String.format(
+			"corpName1=%s & corpName2=%s swap: ",
+			parentSwapStructuresDTO.getCorpName1(),
+			parentSwapStructuresDTO.getCorpName2());
+		try {
+			if (parentSwapStructuresService.swapParentStructures(parentSwapStructuresDTO)) {
+				logger.info(msg + "success");
+				return new ResponseEntity<String>(headers, HttpStatus.OK);
+			} else {
+				logger.info(msg + "failure");
+				return new ResponseEntity<String>(headers, HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Everything looks great and clean so far! I would suggest adding a JSON message in the `ResponseEntity` for both a successful swap or (even better) an error

Reply:
Done. Let me know if the output response looks good or should I add/remove any fields. cc'ing @bffrost too!